dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java

```diff
@@ -30,27 +30,38 @@
  */
 public enum DescriptorLanguage {
     // Add new descriptor language here
-    CWL("CWL", "Common Workflow Language", FileType.DOCKSTORE_CWL, FileType.CWL_TEST_JSON) {
+    CWL("CWL", "Common Workflow Language", FileType.DOCKSTORE_CWL, FileType.CWL_TEST_JSON, false, false) {
         @Override
         public boolean isRelevantFileType(FileType type) {
             return super.isRelevantFileType(type) || type == FileType.DOCKERFILE;
         }
     },
-    WDL("WDL", "Workflow Description Language", FileType.DOCKSTORE_WDL, FileType.WDL_TEST_JSON) {
+    WDL("WDL", "Workflow Description Language", FileType.DOCKSTORE_WDL, FileType.WDL_TEST_JSON, false, false) {
         @Override
         public boolean isRelevantFileType(FileType type) {
             return super.isRelevantFileType(type) || type == FileType.DOCKERFILE;
         }
     },
+    GXFORMAT2("gxformat2", "Galaxy Workflow Format 2", FileType.DOCKSTORE_GXFORMAT2, FileType.GXFORMAT2_TEST_FILE, false, true) {
+        @Override
+        public boolean isRelevantFileType(FileType type) {
+            return super.isRelevantFileType(type);
+        }
+    },
     // DOCKSTORE-2428 - demo how to add new workflow language
-    //SWL("SWL", "Silly Workflow Language", FileType.DOCKSTORE_SWL, FileType.SWL_TEST_JSON)
-    NEXTFLOW("NFL", "Nextflow", FileType.NEXTFLOW_CONFIG, FileType.NEXTFLOW_TEST_PARAMS) {
+    SWL("SWL", "Silly Workflow Language", FileType.DOCKSTORE_SWL, FileType.SWL_TEST_JSON, false, true) {
```

Comment:
Do you want to expose this? It's showing up in the openapi/swagger doc.

Reply:
I think I need to play around with this in a new PR. 
In discussion with @garyluu, I think it can be tricky. For type-checking reasons I think we have to expose every possible language we support in the swagger.yaml/openapi.yaml. 

We can display dynamically at endpoints like https://dockstore.org/api/metadata/descriptorLanguageList a subset of those languages based on what is currently installed (including the languages that are supported by plugins). 

I think we want to update SWL based on feedback from Galaxy, so I'm thinking we keep it.